nova_vm/src/ecmascript/builtins/indexed_collections/typed_array_objects/typed_array_intrinsic_object.rs

```diff
@@ -721,13 +724,103 @@ impl TypedArrayPrototype {
         todo!()
     }
 
+    // ### [23.2.3.16 %TypedArray%.prototype.includes ( searchElement [ , fromIndex ] )](https://tc39.es/ecma262/multipage/indexed-collections.html#sec-%typedarray%.prototype.includes)
+    // The interpretation and use of the arguments of this method are the same as for Array.prototype.includes as defined in 23.1.3.16.
     fn includes(
-        _agent: &mut Agent,
-        _this_value: Value,
-        _: ArgumentsList,
-        _gc: GcScope,
+        agent: &mut Agent,
+        this_value: Value,
+        arguments: ArgumentsList,
+        mut gc: GcScope,
     ) -> JsResult<Value> {
-        todo!()
+        let serach_element = arguments.get(0).bind(gc.nogc());
+        let from_index = arguments.get(1).bind(gc.nogc());
+        // 1. Let O be the this value.
+        let o = this_value;
+        // 2. Let taRecord be ? ValidateTypedArray(O, seq-cst).
+        let ta_record = validate_typed_array(agent, o, Ordering::SeqCst, gc.nogc())?;
+        // 3. Let len be TypedArrayLength(taRecord).
+        let o = TypedArray::try_from(o)
+            .unwrap()
+            .bind(gc.nogc())
+            .scope(agent, gc.nogc());
+        let len = match o.get(agent) {
+            TypedArray::Int8Array(_)
+            | TypedArray::Uint8Array(_)
+            | TypedArray::Uint8ClampedArray(_) => {
+                typed_array_length::<u8>(agent, &ta_record, gc.nogc())
+            }
+            TypedArray::Int16Array(_) | TypedArray::Uint16Array(_) => {
+                typed_array_length::<u16>(agent, &ta_record, gc.nogc())
+            }
+            TypedArray::Int32Array(_)
+            | TypedArray::Uint32Array(_)
+            | TypedArray::Float32Array(_) => {
+                typed_array_length::<u32>(agent, &ta_record, gc.nogc())
+            }
+            TypedArray::BigInt64Array(_)
+            | TypedArray::BigUint64Array(_)
+            | TypedArray::Float64Array(_) => {
+                typed_array_length::<u64>(agent, &ta_record, gc.nogc())
+            }
+        } as i64;
+        // 4. If len = 0, return false.
+        if len == 0 {
+            return Ok(false.into());
+        };
+        // 5. Let n be ? ToIntegerOrInfinity(fromIndex).
+        let n = if let TryResult::Continue(n) =
+            try_to_integer_or_infinity(agent, from_index, gc.nogc())
+        {
+            n?
+        } else {
+            to_integer_or_infinity(agent, from_index, gc.reborrow())?
+        };
```

Comment:
Objects ({}) and Symbols (Symbol) will enter this case. However, since the input is expected to be an integer in most cases, using try should be fine. What do you think?

Reply:
Using the `try` method only really makes sense if you can use it to avoid scoping some or all values. In this case because we need `o` to be scoped for the loop, it doesn't really help any.

So the "try and branch if failed" work is basically not really useful.

Reply:
Thank you, I see. I didn’t fully understand it.
I’ve removed try! 🙇‍♂️

Reply:
Agh! I was wrong! I was somehow thinking this is something like `find` or `forEach` where we always call user code in the loop, but of course not! We're just checking equality!

So your try-branch absolutely made sense, the only thing missing was moving the scoping to be done there on a conditional basis.

I'll push a fix to this branch so you don't have to go back-and-forth with that yourself.